src/views/ContentReferenceWidget.vue

```diff
@@ -66,13 +67,98 @@ export default {
 		},
 	},
 
+	data() {
+		return {
+			element: {
+				...this.richObject,
+				isView: this.richObject.link.includes('view'),
+			},
+		}
+	},
+
+	computed: {
+		tableId() {
+			const id = this.richObject.link.split('/').at(-2)
+			return parseInt(id)
+		},
+		tablePermissions() {
+			return {
+				canCreateRows: this.canCreateRowInElement(this.element),
+				canReadRows: true,
+				canEditRows: this.canUpdateData(this.element),
+				canDeleteRows: this.canDeleteData(this.element),
+				canCreateColumns: false,
+				canEditColumns: false,
+				canDeleteColumns: false,
+				canDeleteTable: false,
+				canSelectRows: false,
+				canHideColumns: false,
+				canFilter: false,
+				showActions: this.canManageElement(this.element),
+			}
+		},
+	},
+
 	mounted() {
 		useResizeObserver(this.$el, (entries) => {
 			const entry = entries[0]
 			const { width } = entry.contentRect
 			this.$el.style.setProperty('--widget-content-width', `${width}px`)
 		})
 	},
+
+	methods: {
+		async createRow() {
+			if (!this.$store) {
```

Comment:
Ah i think my earlier comment I misunderstood what you're doing here. 

We can just register the store in https://github.com/nextcloud/tables/blob/5c8f63e7fabdc3b3d34dc73398479dfd87162032/src/reference.js#L27 instead and avoid all the extra logic as we then just have it available